### PR TITLE
feat(story): add right click on news, record & org

### PR DESCRIPTION
**Pull request by f-necas**
Used to open a in a new tab if router is enabled.

### CODE REVIEW

**Review by fgravin — CHANGES_REQUESTED** [review]
Thanks @f-necas 
It works quite well !
Nevertheless, relying on the `SearchService` implementation is not accurate enough, cause the `SearchSerive` is not a root level service (singleton), but it's instanciated by any search section.

You should check at a higher level "if the router is enabled for this application" IMO.
We should even check is the `DefaultRouterModule` is loaded, but it would make a dependency to the router, which we don't want.

The issue is even more complex, the `feature` components are generic, and saying that the url of an organisation is "a search on the publisher" is a datahub thing, idk, it could land to the organisation description page in another application. This could be handled by overriding the `RouterSearchService.getMetadataRecordTargetUrl()` at the application level though.

 Maybe, just injecting the angular `Router` service as `@optionnal` somewhere could tell if there is a router in the application.

Note that opening a metadata in a new tab is not a "search" thing, it does not trigger any search, so it's abusive to put it in the search service.

Maybe a `RightClickRoutingService` could handle all of these things idk i'm think outloud.

Also, you have to keep in mind that this should work in [the web component](https://geonetwork.github.io/geonetwork-ui/main/demo/webcomponents/gn-results-list.sample.html) as well, where there is already a mechanism to load the metadata in a remote site (not based on href though).

This deserve more thinking  anyway

You are on good track !
Good luck :smirk:

File: apps/datahub/src/app/home/news-page/news-page.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="relative" gnUiSearchStateContainer="newsfeed">
+<div class="relative" gnUiSearchRouterContainer="newsfeed">
```
[the remark this answers is not in the document]
Comment (tkohr) [reply apps/datahub/src/app/home/news-page/news-page.component.html:1]: Yesterday we had a doubt exactly about this point, if two states bound to the router could cause problems. For detecting the presence of a router via the `SearchService` / `RouterSearchService` usage, this seemed like a (almost too) simple solution. Thanks for your input on the alternatives @fgravin !

**Comment by jahow** [comment]
I mean we can also make it much simpler and just have an injection token which is a template of the target url, such as `/geonetwork/blabla/{uuid}`. And then the router module would provide a different value for this injection token. The web component would also provide a different one, and that could also be done at the application level. So no new service, just one injection token (maybe two actually for the orgs).

Like so:
```ts
// somewhere in feature-search
const DATASET_HREF_TEMPLATE = new InjectionToken('dataset-href-template');

// somewhere in feature-catalog
const ORGANIZATION_HREF_TEMPLATE = new InjectionToken('organization-href-template');
```

Parsing a link template is already done in the gn-results-list web component: https://github.com/geonetwork/geonetwork-ui/blob/1688f9a42ad22b0330dc83c5e4faf5eddb3a0684/apps/webcomponents/src/app/components/gn-results-list/gn-results-list.component.ts#L85-L90

File: apps/datahub/src/app/home/news-page/news-page.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (jahow) [reply apps/datahub/src/app/home/news-page/news-page.component.html:1]: Right, that makes sense! having several search states bound to the router is indeed prone to weird behavior at some point. My bad for suggesting this.

**Comment by fgravin** [comment]
> I mean we can also make it much simpler and just have an injection token which is a template of the target url

Yes, an injected string is a great solution indeed. :+1: 
Thanks @jahow for the suggestion.

**Comment by f-necas** [comment]
Thanks all for your help and reviews. 

I finally decided to use an external service to delegate the choice of the route and properties of the object used (record or org) by this service. 

Using an InjectionToken blocked the property of the object to the one we use in the datahub (org.name and record.uuid). I also tried to used it here https://github.com/geonetwork/geonetwork-ui/pull/505/commits/562de2e1985f8f14a00829610d559030ff39e34c but didn't find it very convenient and @tkohr seemed to share the same feeling.

**Review by fgravin — COMMENTED** [review]
Thanks @f-necas , could you argue why you choose the new Service instead of just having InjectionToken as suggested @jahow .
It's the same logic (overriding dependency injection), but less intrusive, as we don't introduce a new service but just tokens (for strings).

**Comment by f-necas** [comment]
I tried to use InjectionToken for this issue but I didn't find it very appropriate by the way I implemented it.

I wanted to delegate to the service the choice of the property used (name, email for the org ? id, uuid or other property for record ?). So, instead of having an InjectionToken just returning a string, it was a function inside it, and that didn't seems a proper way to use it. 
```ts
  getOrganisationUrl(organisation: Organisation): string {
    return `${ROUTER_ROUTE_HOME}/${ROUTER_ROUTE_SEARCH}?${ROUTE_PARAMS.PUBLISHER}=${organisation.name}`
  }
```

I can just pass a fixed InjectionToken which pass `${ROUTER_ROUTE_HOME}/${ROUTER_ROUTE_SEARCH}?${ROUTE_PARAMS.PUBLISHER}=` inside (from datahub app only) without using org as parameter. 

By the way I've found/read today that SearchField used are quite predefined. If it's the case, my choice of using a service is indeed, not appropriate.

**Comment by fgravin** [comment]
Ok I see.
I think that an `InjectionToken` can point to function as well.
Anyway, the balance here is between 
- having one service which aggregate all url functions
- having one InjectionToken per function.

What do you think @jahow ?
I don't have an opinion on this right now.

**Comment by jahow** [comment]
I would still go for injection tokens, we basically only have to support `{uuid}` for records and `{name}` for orgs and that's it. I wouldn't use functions there or even a service, to me it's a case of YAGNI.

**Pull request merged**